pkg/cmd/korectl/configure.go

```
@@ -0,0 +1,231 @@
+				Server:   "local",
+				AuthInfo: "local",
+			},
+		}
+	}
+
+	if config.GetCurrentServer().Endpoint != localEndpoint {
+		config.Servers = map[string]*Server{
+			"local": {Endpoint: localEndpoint},
+		}
+		config.AuthInfos = map[string]*AuthInfo{
+			"local": nil,
+		}
+	}
+}
+
+type authInfoConfig struct {
+	ClientID, ClientSecret string
+	AuthorizeURL           string
+	prompts                prompts
+}
+
+func newAuthInfoConfig(config *Config) *authInfoConfig {
+	result := &authInfoConfig{}
+
+	if config.AuthInfos["local"] != nil {
+		result.ClientID = config.GetCurrentAuthInfo().OIDC.ClientID
+		result.ClientSecret = config.GetCurrentAuthInfo().OIDC.ClientSecret
+		result.AuthorizeURL = config.GetCurrentAuthInfo().OIDC.AuthorizeURL
+	}
+
+	return result
+}
+
+func (a *authInfoConfig) createPrompts() *authInfoConfig {
+	a.prompts = prompts{
+		&prompt{id: "Client ID", errMsg: "%s cannot be blank", value: a.ClientID},
+		&prompt{id: "Client Secret", errMsg: "%s cannot be blank", value: a.ClientSecret},
+		&prompt{id: "OpenID endpoint", errMsg: "%s cannot be blank", value: a.AuthorizeURL},
+	}
+	return a
+}
+
+func (a *authInfoConfig) collectPrompts() (*authInfoConfig, error) {
+	if err := a.prompts.collect(); err != nil {
+		return nil, err
+	}
+	a.ClientID = a.prompts.getValue("Client ID")
+	a.ClientSecret = a.prompts.getValue("Client Secret")
+	a.AuthorizeURL = a.prompts.getValue("OpenID endpoint")
+	return a, nil
+}
+
+func (a *authInfoConfig) update(config *Config) {
+	config.AuthInfos = map[string]*AuthInfo{
+		"local": {
+			Token: nil,
+			OIDC: &OIDC{
+				ClientID:     a.ClientID,
+				ClientSecret: a.ClientSecret,
+				AuthorizeURL: a.AuthorizeURL,
+			},
+		},
+	}
+}
+
+type gcpInfoConfig struct {
+	Region  string
+	Project string
+	KeyPath string
+	prompts prompts
+}
+
+func newGcpInfoConfig() *gcpInfoConfig {
+	return &gcpInfoConfig{}
+}
+
+func (g *gcpInfoConfig) load(src string) *gcpInfoConfig {
+	content, err := ioutil.ReadFile(src)
```

All errors are swallowed in load, we should not do that.

---

After sleeping on this: the argument makes sense.